### PR TITLE
feat(core): facelift

### PR DESCRIPTION
**Pull request by robinpyon**
# Key objectives

- Refresh our studio colour palette and bring it inline with modern tooling
- Simplify and reduce elements where possible – mask the complexity inherent in our current pane navigation paradigm
- A focus on adding *calmness* to our interface – not only by reducing unnecessary elements – but also other micro-optimisations that can limit what’s on screen (delayed tooltips, animated dialogs and field actions)
- Carrying on from this, making the document editor feel a little less crowded and making document panes feel like user *destinations*
- A huge focus on consistency – not only in styling but also interactivity too. Having multiple lockups / variations of common UI elements makes our interface feel unstable.
- Provide simple and opinionated components for our internal developers that can help reduce the amount of layout boilerplate

Screenshot 2023-12-14 at 17 00 03

# Notable changes 
## Updated default colour themes
These new themes for both light and dark modes provide added vibrancy to existing tones. 

Application of these colours within the studio are indexed towards reducing visual noise – enforcing key buttons and editable surfaces stand out, whilst visual dividers recede further into the background.

### New major version of Sanity UI (to be released)
- makes minor visual tweaks to our core primitives and components
- provides more sensible defaults
- adds delight to some core components with simple animation
- overhauls our current theming engine and paves the way for a simpler theming API surface in future

### Updated Studio Typography
- Inter font will be the new default font.

## New Workspace menu / home button
We now clearly signpost the current workspace a user is in – critical for larger studios.
As a result of this change, users will no longer be able to specify custom logos that always appear independently of the current workspace
## Sanity branding for simpler studios without customisation
Studios with a single workspace will now show a Sanity logo by default
Screenshot 2023-12-14 at 17 03 48

### Updated navigation

A compact preview style has been added which not only reduces visual noise when browsing document lists, but also helps users differentiate between documents and structure folders

Screenshot 2023-12-14 at 17 05 40

### Updated form document layout

- Document forms now enjoy more spacing and have a dedicated large title (which uses the document preview value) to make these panes stand out more as *destinations* for editors.
- The footer has been simplified with more human readable dates to better indicate published status
Screenshot 2023-12-14 at 17 06 45

# Studio UI Components

## Highly opinionated components packed with most common use cases 💥 
Highly opinionated Sanity UI components with restrict many (if not all) styling props, and help avoid often-repeated boilerplate in common scenarios and force consistency.
Currently located in `packages/sanity/src/ui/*`

- [Button](https://github.com/sanity-io/sanity/pull/4927/files#diff-3f7191473443a269de1b2cea6ad6ebecc06eb618691e7d4fea1fde4de30ae1c0) 
- [Dialog](https://github.com/sanity-io/sanity/pull/4927/files#diff-084498acbdf909601ca715e3fac34257430fab5647b596090aa51d8b1d846603)
- [MenuItem](https://github.com/sanity-io/sanity/pull/4927/files#diff-aefb2071fd91935f9d50f29267fa1f2fc9f477f3c77c842c57e346714fd5ee34)
- [MenuGroup](https://github.com/sanity-io/sanity/pull/4927/files#diff-04b1aebb0d0827f844300970e6745822ee62e86d1bb6cdcf3e357e8dcbb48a02)
- [Popover](https://github.com/sanity-io/sanity/pull/4927/files#diff-4f91b6cc4b1d92ebebae910563f4658fbc5598d7fe3461b25556baba8e2eec64)
- [Tab](https://github.com/sanity-io/sanity/pull/4927/files#diff-be07526872eec424ca3611a328f0c5186edfd1569ec431dda7f6fe95cd9f58e6)
- [Tooltip](https://github.com/sanity-io/sanity/pull/4927/files#diff-8ad193cfd30ca1e45657badfe83ad1ff41d71be2a8edb942fa120c505a4207b1)

**Pre and post of a button with a tooltip**
Screenshot 2023-12-14 at 17 09 55

## no-restricted-imports 
Encourages import of opinionated versions of the “full flat” sanity ui components. Enforcing consistency and ease of use

Screenshot 2023-12-14 at 17 10 57

## Common components
Commonly used components have been abstracted out to aid with consistency (and simplify i18n translation)
- [Loading component (Text + Spinner lockup)](https://github.com/sanity-io/sanity/pull/4927/files#diff-e2d699aa52b49a7b55b5f69967485b6d394c8fa7f93d64fa956554a04eaedc61)
   Out of the box delay and animation.
- [Contextual menu buttons](https://github.com/sanity-io/sanity/pull/4927/files#diff-6542127768fb7c1de4345c02adebae8472889a926c563c7080a481d670c8bebb)
   Out of the box icon and tooltip props.
-[ Unified document status (single dot representation)](https://github.com/sanity-io/sanity/pull/4927/files#diff-b8fd5b35e7924d909adcd618824a99e048f4e2735f6ea39b3f89ead37d3549d8)
   Unified representation, for updated dates and status

# Deprecations
## Studio configuration
`studio.components.logo`
Custom logo components are no longer rendered.
Users are encouraged to provide custom components for individual workspace icons instead.

## Tool API
`icon`
Custom tool icons are no longer rendered.

## Structure builder
`menuItemGroup.title`
Menu item groups are no longer displayed, but these can still be defined.
## [I18n keys ](https://github.com/sanity-io/sanity/pull/4927/files#diff-5c6b73dd5c1114ae88bf772a5e5f8cb88d3b696adfed0df4bc2e450c6ada078f)

# Migration
- This work will land in a minor studio release and major Sanity UI release
- Users with custom logos who wish to continue displaying them will need to reformat them to fit within workspace icons.
- Third party plugins will need to update to Sanity UI v2 in order to take advantage of updated styles.
- Those which don’t will continue to work, but will look out of place until updated

### CODE REVIEW

File: packages/sanity/src/desk/comments/src/components/avatars/SpacerAvatar.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'
+import {studioTheme} from '@sanity/ui'
 
-export const AVATAR_HEIGHT = 25
+export const AVATAR_HEIGHT = studioTheme.avatar.sizes[0].size
```
Comment (hermanwikner) [inline packages/sanity/src/desk/comments/src/components/avatars/SpacerAvatar.tsx:4]: This value should be `25` (the same width as the avatars used in comments) for the spacing to be correct in the UI. The values from `studioTheme` for avatar sizes do not have 25, so this value must be hardcoded here to 25.

Screenshot 2023-12-19 at 08 30 21

File: dev/test-studio/structure/resolveStructure.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ import {delayValue} from './_helpers'
 import {typesInOptionGroup} from './groupByOption'
 
 export const structure: StructureResolver = (S, {schema, documentStore, i18n}) => {
+  i18n.loadNamespaces(['testStudio']) // @todo: confirm – is this idiomatic usage?
```
Comment (pedrobonamin) [inline dev/test-studio/structure/resolveStructure.ts:36]: Do you know if this necessary here? 
cc: @bjoerge @rexxars

Comment (bjoerge) [reply dev/test-studio/structure/resolveStructure.ts:36]: If it's been added in this branch I would assume it's not, and maybe introduced by a merge error?

File: packages/sanity/src/core/form/inputs/InvalidValueInput/UntypedValueInput.tsx [hunk 3]
```diff
@@ -34,17 +35,9 @@ function SetMissingTypeButton({
   return (
     <Button
       onClick={handleClick}
-      text={
-        <>
-          <Translate
```
Comment (pedrobonamin) [inline packages/sanity/src/core/form/inputs/InvalidValueInput/UntypedValueInput.tsx:39]: What's the difference in using  `<Translate />` vs `t()` ? Button is expecting a string in text and if we use the `<Translate /> ` it return a node

File: packages/sanity/src/desk/comments/src/components/pte/blocks/MentionInlineBlock.tsx [hunk 4]
```diff
@@ -40,10 +39,8 @@ export function MentionInlineBlock(props: MentionInlineBlockProps) {
   return (
     <Tooltip
       portal
-      open={selected}
```
Comment (pedrobonamin) [inline packages/sanity/src/desk/comments/src/components/pte/blocks/MentionInlineBlock.tsx:43]: @hermanwikner could you please help me confirm this is not changing any previous behaviour? Couldn't find a way to reproduce this "selected" .
Thanks

File: packages/sanity/src/ui-components/tooltip/constants.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const TOOLTIP_DELAY_PROPS = {
```
Comment (pedrobonamin) [inline packages/sanity/src/ui-components/tooltip/constants.ts:1]: I think we could consider exporting a TooltipDelayGroupProvider with this default prop as fast follow.

File: test/e2e/tests/components-api/studio.spec.ts [hunk 6]
```diff
@@ -21,16 +21,6 @@ test.describe('Studio Components API:', () => {
     ).toBeVisible()
   })
 
-  test('studio.components.logo', async ({page}) => {
```
Comment (pedrobonamin) [inline test/e2e/tests/components-api/studio.spec.ts:24]: This was removed, so this test won't apply anymore

**Review by pedrobonamin — APPROVED** [review]

File: packages/sanity/src/core/components/documentStatus/DocumentStatus.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        <StyledText size={1} weight="medium">
+          Published {publishedDate}
+        </StyledText>
+      )}
+      {updatedDate && (
+        // eslint-disable-next-line i18next/no-literal-string
+        <StyledText muted size={1} wrap="nowrap">
+          Edited {updatedDate}
+        </StyledText>
+      )}
```
Comment (rexxars) [inline packages/sanity/src/core/components/documentStatus/DocumentStatus.tsx:77]: These need to be localized

**Review by rexxars — CHANGES_REQUESTED** [review]

File: packages/sanity/src/core/components/documentStatus/DocumentStatus.tsx (hunk 7 above)
Comment (pedrobonamin) [reply packages/sanity/src/core/components/documentStatus/DocumentStatus.tsx]: Updated! Thanks

**Review by rexxars — APPROVED** [review]

**Pull request merged**